src/Preg.php

```diff
@@ -45,7 +45,7 @@ public static function match($pattern, $subject, &$matches = null, $flags = 0, $
             return $result;
         }
 
-        throw new PregException('Error occurred when calling preg_match.', preg_last_error());
+        throw self::newPregException(preg_last_error(), __METHOD__, (array) $pattern);
```

Comment:
We can and I think we also do. However, this call is necessary to obtain the original last error of the preg operation as testing a patterin inside that routine will overwrite it and this parameter is the default exception code which otherwise would be implicit only and at risk to get overwritten due to more preg operations in the subroutine.

Please let me know if this answers your question and if not, please help me to better understand the intend of your question.